app/src/androidTest/java/org/mozilla/focus/activity/ClearCacheTest.java

```diff
@@ -0,0 +1,66 @@
+package org.mozilla.focus.activity;
+
+import android.content.Intent;
+import android.support.annotation.Keep;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mozilla.focus.R;
+import org.mozilla.focus.utils.AndroidTestUtils;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.RootMatchers.withDecorView;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+@Keep
+@RunWith(AndroidJUnit4.class)
+public class ClearCacheTest {
+
+    // Defer the startup of the activity cause we want to avoid First Run / Share App / Rate App dialogs
+    @Rule
+    public final ActivityTestRule<MainActivity> activityRule = new ActivityTestRule<>(MainActivity.class, true, false);
+
+    @Before
+    public void setUp() {
+        // Set the share preferences and start the activity
+        AndroidTestUtils.beforeTest();
+    }
+
+    /**
+     * Test case no: TC0044
+     * Test case name: Clear cache
+     * Steps:
+     * 1. Launch rocket
+     * 2. Tap menu
+     * 3. Tap clear cache
+     * 4. Check toast message "cache cleared"
+     */
+    @Test
+    public void clearCache() {
+
+        // Now start the activity
+        activityRule.launchActivity(new Intent());
+
+        // Tap menu
+        AndroidTestUtils.tapBrowserMenuButton();
+
+        // Tap clear cache
+        onView(withId(R.id.menu_delete)).perform(click());
+
+        // Check toast message "cache cleared"
+        onView(withText(containsString("cache cleared")))
```

Comment:
use string resource instead of hardcoded string.

Reply:
why not use  AndroidTestUtils.toastContainsText?

Reply:
since AndroidTestUtils.toastContainsText applying withTest() will assert string mismatch "%1$s cache cleared", after off-line discussion, I propose to use AndroidTestUtils.removeStrFormatter() which will remove formatter based on regex pattern and we will go with it.